gapil/resolver/statement.go

```diff
@@ -172,6 +174,15 @@ func delete_(rv *resolver, in *ast.Delete) *semantic.MapRemove {
 	return &semantic.MapRemove{AST: in, Type: mty, Map: m, Key: k}
 }
 
+func clear_(rv *resolver, in *ast.Clear) *semantic.MapClear {
+	m := expression(rv, in.Map)
+	mty, ok := m.ExpressionType().(*semantic.Map)
+	if !ok {
+		rv.errorf(in.Map, "clear's argument must be a map, got %s", typename(m.ExpressionType()))
+	}
+	return &semantic.MapClear{AST: in, Type: mty, Map: m}
```

Comment:
You are missing a mapping between the `ast` and `semantic` nodes here:

```golang
out := &semantic.MapClear{AST: in, Type: mty, Map: m}
rv.mappings.Add(in, out)
return out
```

Without this, you won't get debug info generated for this statement.